src/swarm-tui/Swarm/TUI/View.hs

```diff
@@ -1276,15 +1276,19 @@ explainCapabilities gs e
       ]
 
   renderCmdInfo c =
-    padTop (Pad 1) $
-      vBox
-        [ hBox
-            [ padRight (Pad 1) (txt . syntax $ constInfo c)
-            , padRight (Pad 1) (txt ":")
-            , withAttr magentaAttr . txt . prettyText $ inferConst c
-            ]
-        , padTop (Pad 1) . padLeft (Pad 1) . txtWrap . briefDoc . constDoc $ constInfo c
-        ]
+    Widget Fixed Fixed $ do
+      ctx <- getContext
+      let w = ctx ^. availWidthL
+      render
+        . padTop (Pad 1)
+        $ vBox
+          [ hBox
+              [ padRight (Pad 1) (txt . syntax $ constInfo c)
+              , padRight (Pad 1) (txt ":")
+              , withAttr magentaAttr . txt $ prettyTextWidth (inferConst c) w
```

Comment:
Using the width `w` of the widget to pretty-print the type signature here is not quite right, because `w` is the width of the *entire* widget, but the type signature is not going to be printed starting in the first column of the widget; there is a command name and colon that come before it. You can see the issue in the following screenshots.  In the first screenshot, you can see that the type of `installKeyHandler` is cut off, because the pretty-printer thought it had the entire width of the widget to work with, so it laid out the type on a single line:

![prettynowrap](https://github.com/swarm-game/swarm/assets/533859/ce36a070-ea1b-42d7-9626-664da7f1ca01)

In the next screenshot, you can see that in a much narrower window, the pretty-printer does actually insert some newlines into the type, but it still gets cut off on the right:

![prettywrap](https://github.com/swarm-game/swarm/assets/533859/16d9a38c-95d9-4654-a823-b2f68fc73404)

I can imagine two things to do here.
- The obvious thing would just be to subtract the width of the constant name + colon (+ two spaces) from `w` before passing it to `prettyTextWidth`.
- The nicer, slightly more complex approach would be:
    1. first try printing the type on a single line with `prettyTextLine` and see whether the command name, colon, + pretty-printed type will fit within `w`; if so, great.
    2. If not, instead use `prettyTextWidth` to print the type with a width of `w - 2` and then print the type indented by 2 spaces on a new line(s) after the command + colon.

Reply:
Update on this:
I managed to get rendering a little bit better but need to test the implemented a bit more. Here's how the above screenshot renders now:
![CleanShot 2024-06-14 at 00 46 32@2x](https://github.com/swarm-game/swarm/assets/15181803/7dfbe752-4086-4608-a09e-d7f9de0663df)

Hopefully this is what you had in mind @byorgey ?

Reply:
Yes, that looks nice!